### PR TITLE
Update release documentation

### PR DESCRIPTION
**Pull request by SuperQ**
Update the release documentation about dependencies.
* We now have dependabot to auto-update things.
* Add note about some manual dependency work.
* Add note about how to check for outdated React code.

Signed-off-by: SuperQ <superq@gmail.com>

### CODE REVIEW

File: RELEASE.md
Comment (beorn7) [inline RELEASE.md:80]: Perhaps express somehow that this is a continuous process, so that when it comes to a release, most work is already done.

As is, it could be understood that the release shepherd has to push a button to initiate the automatic update or something.

Suggestion:

Note that we use [Dependabot](.github/dependabot.yml) to continuously update most things automatically. Therefore, mest dependencies should be up to date already. Check the [dependencies GitHub label](https://github.com/prometheus/prometheus/labels/dependencies) to see if there are any pending updates.

File: RELEASE.md
Comment (beorn7) [inline RELEASE.md:93]: Perhaps add a note along the lines: This is usually only required for dependencies not following the Go Modules semver semantics.

**Review by beorn7 — CHANGES_REQUESTED** [review]
Excellent. Thanks a lot.

Just a few clarifying suggestions.

**Comment by Nexucis** [comment]
just for the record, I deactivated dependabot for the frontend, because it was too noisy. I'm looking for an alternative solution that would open a single PR to update the dependency.

dependabot is still used for the backend because it's "manageable"

File: RELEASE.md
Comment (SuperQ) [reply RELEASE.md:93]: I added a note about this in the dependabot section, but I'll add it here as well.

File: RELEASE.md
Comment (SuperQ) [reply RELEASE.md]: Yes, that's better. Done.

**Comment by Nexucis** [comment]
@SuperQ yeah I know, but none can say to dependabot to create a single PR. I think that would really help to maintain a frontend application.
The frequency is not really the issue, it's more the number of the PRs opened by dependabot for the react-app (which is super huge).

File: RELEASE.md
Comment (beorn7) [inline RELEASE.md:99]: An "is" is missing.

**Review by beorn7 — CHANGES_REQUESTED** [review]
WRT React: For the time being, perhaps we should explain here that React updates are done manually, and that release shepherds that are not React experts should talk to @Nexucis and @juliusv to find out if there are updates necessary for the release?

**Comment by Nexucis** [comment]
with @SuperQ we re-enabled dependabot for the frontend with a monthly frequency. Let's see if with this configuration it's less painful to manage.
And like that when it's the release date, then the dependencies for backend or frontend will be up to date.

**Comment by Nexucis** [comment]
And of course @beorn7 , for "safety" or peaceful mind, we can put my name (and certainly @juliusv if he is fine too) in case there is an issue.
But it's more or less the purpose of the Code owner configuration. So don't know if it really makes sense to add these point of contacts again.

**Comment by beorn7** [comment]
Mostly I think it's important to clearly communicate that updating the React dependencies is not primarily a responsibility of the release shepherd.

**Comment by Nexucis** [comment]
yeah @beorn7 at least on my side, that's totally the idea when we enabled dependabot: that the upgrade of the dependency is not anymore the duty of release shepherd.
But now we are talking about this thought, I'm wondering if it's a shared thought or if it's obvious and I'm overthinking.

File: RELEASE.md
Comment (Nexucis) [reply RELEASE.md:116]: I tackled this in #10517.
And actually `npm outdated` works as well.

I'm wondering what is the best between `npm outdated` and the other way I choose.

**Comment by Nexucis** [comment]
@SuperQ I think you can remove your changes regarding the UI part. As I said i'm tackling these issues in #10517.

And i think regarding the configuration of dependabot for the UI side, I think I will deactivate it. @juliusv said he would prefer the manual way in #9973 and I would agree on that.

Perhaps once #10517 is merged, it will be easier to merge the dependabot PR's for the UI.

File: RELEASE.md
Comment (SuperQ) [reply RELEASE.md]: Ok, I reverted this section.

**Review by Nexucis — APPROVED** [review]

File: RELEASE.md
Comment (beorn7) [inline RELEASE.md:83]: A "to" is missing here somewhere. How about "Note that we use [Dependabot](.github/dependabot.yml) to continuously update…"?

**Review by beorn7 — APPROVED** [review]
Modulo nit about a missing word.

**Review by beorn7 — APPROVED** [review]

**Pull request merged**
